frontend/packages/console-dynamic-plugin-sdk/src/extensions/yaml-templates.ts

```diff
@@ -0,0 +1,24 @@
+import { Extension } from '@console/plugin-sdk/src/typings/base';
+import { K8sKind } from 'public/module/k8s/types';
+
+export namespace ExtensionProperties {
+  export type YAMLTemplate = {
+    /** Model associated with the template. */
+    model: K8sKind;
+    /** The YAML template to use. */
+    template: string;
+    /** The name of the template. If not specified, use the `default` value. */
+    templateName?: string;
```

Comment:
Simplify to `name`.
I prefer to force the user to mark the template as the `default` instead of simply omitting the name.
```suggestion
    /** The name of the template. Use the name `default` to mark this as the default template. */
    name: string | 'default';
```